Automattic/woocommerce-payments
# Add compatibility between Multi-Currency and WCPay Subscriptions.

Fixes #3063

#### Changes proposed in this Pull Request

As far as I can tell, WCPay Subs uses the same filters, etc that the main Subs plugin uses, so there wasn't much to be done. 

* One additional check in `should_convert_product_price` to _not_ convert the product price during an internal check within WCPay Subs.
  *  The check in Subs is here (linked to PR since code may move in develop): https://github.com/Automattic/woocommerce-payments/pull/3095/files#diff-61d838bb73eba5daad8c16c1a6b535141ec17a004f0cf7e577b522c20f98d2b3R733
  * We need to make sure that the product price and the subscription price do not match so that we send the proper price and currency to Stripe.

#### Testing instructions

* Create a new subscription product with a signup fee.
* Add the sub to your cart.
* Make sure that the sub and signup fees are converted correctly.
* Purchase the subscription.

Note: A customer in your store can only have one currency for subscriptions, so if you make a purchase in CAD, that customer will only be able to sign up again with CAD. If you try another currency you will get an error logged.

-------------------


**Post merge**



- [x] https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-3.2.0---Subscriptions/#compatibility-with-multi-currency
